ugs-core/src/com/willwinder/universalgcodesender/firmware/fluidnc/FluidNCSettings.java

```diff
@@ -145,12 +158,68 @@ public List<FirmwareSetting> getAllSettings() {
 
     @Override
     public boolean isHardLimitsEnabled() {
-        return false;
+        return getSetting("axes/x/motor0/hard_limits").filter(s -> StringUtils.equalsIgnoreCase("true", s.getValue())).isPresent() ||
+                getSetting("axes/x/motor1/hard_limits").filter(s -> StringUtils.equalsIgnoreCase("true", s.getValue())).isPresent() ||
+                getSetting("axes/y/motor0/hard_limits").filter(s -> StringUtils.equalsIgnoreCase("true", s.getValue())).isPresent() ||
+                getSetting("axes/y/motor1/hard_limits").filter(s -> StringUtils.equalsIgnoreCase("true", s.getValue())).isPresent() ||
+                getSetting("axes/z/motor0/hard_limits").filter(s -> StringUtils.equalsIgnoreCase("true", s.getValue())).isPresent() ||
+                getSetting("axes/z/motor1/hard_limits").filter(s -> StringUtils.equalsIgnoreCase("true", s.getValue())).isPresent();
     }
-
+    
+    private boolean checkMotorHasEndstop( String aAxis, String aMotor) {
+        return getSetting("axes/"+aAxis+"/"+aMotor+"/limit_neg_pin").filter(s -> !StringUtils.equalsIgnoreCase("NO_PIN", s.getValue())).isPresent() || 
+                getSetting("axes/"+aAxis+"/"+aMotor+"/limit_pos_pin").filter(s -> !StringUtils.equalsIgnoreCase("NO_PIN", s.getValue())).isPresent() || 
+                getSetting("axes/"+aAxis+"/"+aMotor+"/limit_all_pin").filter(s -> !StringUtils.equalsIgnoreCase("NO_PIN", s.getValue())).isPresent();
+    }
+    
+    @Override
+    public boolean hasX0() {
+        return checkMotorHasEndstop("x","motor0");
+    }
+    
+    @Override
+    public boolean hasX1() {
+        return checkMotorHasEndstop("x","motor1");
+    }
+    
+    @Override
+    public boolean hasY0() {
+        return checkMotorHasEndstop("y","motor0");
+    }
+    
+    @Override
+    public boolean hasY1() {
+        return checkMotorHasEndstop("y","motor1");
+    }
+    
+    @Override
+    public boolean hasZ0() {
+        return checkMotorHasEndstop("z","motor0");
+    }
+    
+    @Override
+    public boolean hasZ1() {
+        return checkMotorHasEndstop("z","motor1");
+    }    
+    private void setHardLimitEnabled(String aAxis,String aMotor,boolean enabled) throws FirmwareSettingsException{
```

Comment:
Even if this is private I'd prefer to use the Axis enum instead of magic strings.